docs/source/workflow/generate-types.mdx

```diff
@@ -13,47 +13,56 @@ You can use these generated TS types in your resolvers to type-check that your r
 
 ## Setting up your project
 
-We'll use the [GraphQL Code Generator](https://www.the-guild.dev/graphql/codegen) library to generate types based on our GraphQL schema. There are [multiple ways](https://www.the-guild.dev/graphql/codegen/plugins/typescript/typescript-resolvers#usage-example) to provide a schema to GraphQL Code Generator. Below, we'll show one of the common methods, beginning by wrapping our schema in a `gql` template literal tag from the [`graphql-tag`](https://www.npmjs.com/package/graphql-tag) package.
+We'll use the [GraphQL Code Generator](https://www.the-guild.dev/graphql/codegen) library to generate types based on our GraphQL schema. There are [multiple ways](https://www.the-guild.dev/graphql/codegen/plugins/typescript/typescript-resolvers#usage-example) to provide a schema to GraphQL Code Generator. Below, we'll show the most common method, which requires our schema to be in a `.graphql` file.
 
-If you haven't already, begin by installing the `graphql-tag` library:
+If you haven't already, move your server's schema into a `.graphql` file, like so:
 
-```bash
-npm install graphql-tag
-```
-
-The `gql` template literal tag from `graphql-tag` can parse a Schema Definition Language (SDL) string into an abstract syntax tree (AST). Apollo Server accepts a string or an AST as a valid option for defining a schema.  
+```graphql title="schema.graphql"
+type Query {
+  books: [Book]
+}
 
-If we pass GraphQL Code Generator a TypeScript file path, it will automatically generate types for the `gql` tag within that file. So, we can wrap our schema string in a `gql` tag to let GraphQL Code Generator know it should generate types for that tag. 
+type Book {
+  title: String
+  author: String
+}
 
-Import the `gql` tag into the file where you define your schema, then wrap your schema string like so:
+type AddBookMutationResponse {
+  code: String!
+  success: Boolean!
+  message: String!
+  book: Book
+}
 
-```ts title="src/schema.ts"
-import gql from 'graphql-tag';
+type Mutation {
+  addBook(title: String, author: String): AddBookMutationResponse
+}
+```
 
-export default gql`
-  type Query {
-    books: [Book]
-  }
+If you moved your schema into a `.graphql` file, update your imports to ensure you're still properly passing your schema to your server. In the file where you create your server, you can read in your schema using `readFileSync` from the `fs` package:
 
-  type Book {
-    title: String
-    author: String
-  }
+```ts title="src/index.ts"
+// ...other imports
+import { readFileSync } from 'fs';
+import path, { dirname } from 'path';
+import { fileURLToPath } from 'url';
+
+// If you are using ESM module syntax you can redefine `__dirname` as 
```

Comment:
The other alternative is to just use a relative path and rely on the program being run with the root directory of the project as the working directory.  When you use `npm run` I believe that's the case. That ends up a bit simpler though you at least want to be clear that it's relying on being run from `npm run`. Thoughts @trevor-scheer ?

Reply:
Yeah, I'm onboard with that change. I do think it's nice for code examples to be copy/pasteable and it just works. As it is currently, CJS users would expect a runtime error without removing that line.

Reply:
Cool, changed this to:

```ts
// ...other imports
import { readFileSync } from 'fs';

// Note: this only works locally because it relies on `npm` routing
// from the root directory of the project.
const typeDefs = readFileSync('./schema.graphql', { encoding: 'utf-8' });
```